ktor-client/ktor-client-features/ktor-client-json/ktor-client-serialization/common/src/io/ktor/client/features/json/serializer/KotlinxSerializer.kt

```diff
@@ -28,34 +28,34 @@ class KotlinxSerializer(
     }
 
     internal fun writeContent(data: Any): String =
-        json.stringify(buildSerializer(data, json.context) as KSerializer<Any>, data)
+        json.encodeToString(buildSerializer(data, json.serializersModule) as KSerializer<Any>, data)
 
     override fun read(type: TypeInfo, body: Input): Any {
         val text = body.readText()
-        val deserializationStrategy = json.context.getContextual(type.type)
+        val deserializationStrategy = json.serializersModule.getContextual(type.type)
         val mapper = deserializationStrategy ?: (type.kotlinType?.let { serializer(it) } ?: type.type.serializer())
-        return json.parse(mapper, text)!!
+        return json.decodeFromString(mapper, text)!!
     }
 
     companion object {
         /**
          * Default [Json] configuration for [KotlinxSerializer].
          */
-        val DefaultJsonConfiguration: JsonConfiguration = JsonConfiguration(
-            isLenient = false,
-            ignoreUnknownKeys = false,
-            serializeSpecialFloatingPointValues = true,
+        val DefaultJsonConfiguration: Json = Json {
```

Comment:
A little bit confusing since this is no longer a configuration but Json instance itself

Reply:
Do you have an idea of providing source-compatible change here? The configuration is deprecated.

Reply:
Added deprecations